Springer Books.js

```diff
@@ -0,0 +1,293 @@
+{
+	"translatorID": "b38a44f4-b8af-4553-9edf-fba4d2598d6a",
+	"label": "Springer Books",
+	"creator": "Jonathan Schulz",
+	"target": "^https?://www\\.springer\\.com/\\w\\w/(book|search)\\W",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 12,
+	"browserSupport": "gcsbv",
+	"lastUpdated": "2017-10-23 17:56:11"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2017 YourName
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	var action = url.match(/^https?:\/\/[^\/]+\/[^\/]+\/([^\/?#]+)/);
+	if(!action) return;
+	switch(action[1]) {
+		case "book":
+			// test if any relevant <meta> information is available
+			if(ZU.xpathText(doc, '//meta[@property="og:title"]/@content')) return "book";
+			//test for relevant search entries
+		case "search": return getSearchResults(doc, true) ? "multiple" : false;
+	}
+}
+
+function doWeb(doc, url) {
+	var detection_type = detectWeb(doc, url);
+	switch (detection_type) {
+		case "book": scrapeBook(doc, url);
+		case "multiple": {
+			if (!getSearchResults(doc, true)) return;
+			Zotero.selectItems(getSearchResults(doc, false), function (items) {
+				if (!items) return true;
+				var articles = [];
+				for (var i in items) {
+					articles.push(i);
+				}
+				ZU.processDocuments(articles, scrapeBook);
+			});
+		}
+	}
+	
+}
+
+//returns true if origStr is a shorter version of newStr
+function shortenedStringTest(origStr, newStr) {
+	if (origStr.length > newStr.length) return false;
+	if (origStr.length < 10)
+		return (origStr == newStr.substring(0, origStr.length));
+	else
+		return (origStr.substring(0,9) == newStr.substring(0,9));
```

Comment:
I don't understand this function. The `9` looks like an arbitrary number here. Can you clarify?

Reply:
This function is a plausibility check if one string is a shortened version of a different one, e.g. `Quantum Mechanics - Methods and ...` and `Quantum Mechanics - Methods and Applications`. It essentially compares the first 10 characters, or less if the string is shorter. See below why this is needed. One could, alternatively, check the first half of both strings; checking the full string fails in some cases where a "..." appears in the shortened string.